ambry-tools/src/main/java/com.github.ambry/store/DumpIndexTool.java

```diff
@@ -309,29 +369,50 @@ public String toString() {
     final Set<String> filterSet;
 
     /**
-     * The number of index entries to process every second.
+     * The number of index entries to process every second. In case of {@link Operation#VerifyDataNode}, this is across
+     * all partitions being processed.
      */
     @Config("index.entries.to.process.per.sec")
     @Default("Long.MAX_VALUE")
     final long indexEntriesToProcessPerSec;
 
+    /**
+     * Used for {@link Operation#VerifyIndex} and {@link Operation#VerifyDataNode} if the presence of crafted IDs needs
+     * to be reported as a failure. This check takes precendence over index sanity.
+     */
+    @Config("fail.if.crafted.ids.present")
+    @Default("false")
+    final boolean failIfCraftedIdsPresent;
```

Comment:
Maybe rename this to `reportFailureIfCraftedIdsPresent`? Same for Config (`report.failure.if.crafted.ids.present`)

Reply:
The script also "fails" actually - the exit code returned is not 0.